include/eve/module/core/regular/impl/simd/arm/sve/abs.hpp

```diff
@@ -0,0 +1,50 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/arch/arm/sve/sve_true.hpp>
+#include <eve/concept/value.hpp>
+#include <eve/detail/category.hpp>
+#include <eve/detail/implementation.hpp>
+
+namespace eve::detail
+{
+template<scalar_value T, typename N>
+EVE_FORCEINLINE auto
+abs_(EVE_SUPPORTS(sve_), wide<T, N> const& a) noexcept -> wide<T, N>
+requires sve_abi<abi_t<T, N>>
+{
+  constexpr auto c = categorize<wide<T, N>>();
+  using enum detail::category;
+
+  if constexpr( match(c, unsigned_) ) return a;
+  else                                return svabs_z(sve_true<T>(),a);
+}
+
+template<conditional_expr C, scalar_value T, typename N>
+EVE_FORCEINLINE auto
+abs_(EVE_SUPPORTS(sve_), C const& cx, wide<T, N> const& v) noexcept -> wide<T, N>
+requires sve_abi<abi_t<T, N>>
+{
+  constexpr auto c = categorize<wide<T, N>>();
+  using enum detail::category;
+
+  if constexpr( match(c, unsigned_) ) return v;
+  else
+  {
+    if constexpr( C::is_complete ) return abs_(EVE_RETARGET(cpu_), cx, v);
+    else
+    {
+      auto src = alternative(cx, v, as<wide<T, N>> {});
+      auto m   = expand_mask(cx, as<wide<T, N>> {});
```

Comment:
what's alternative and expand mask? To me seems like if it's mothing - you do wth true, if it's a logical - you convert and  pass it, if it's relative - to logical and pass it. No?

Reply:
that's old code from when we factorized conditional and conditional processing so that we don't have to have a bunch of if constexpr in every conditional support.

Reply:
Ah - found it - ok.